docs: Workflow of workflows pattern (#3536)
argoproj/argo: docs/workflow-of-workflows.md

# Workflow of Workflows

> v2.9 and after

## Introduction

> A workflow can trigger one or more workflow and manager their result.

Reviewer: ```suggestion
The Workflow of Workflows pattern involves a parent workflow triggering one or more child workflows, managing them, and acting their results.
```

# Workflow of Workflows

> v2.9 and after

> ## usecase

Reviewer: ```suggestion
## Introduction
```

> Workflow needs to trigger one of more workflows and manage it. 

Reviewer: A workflow can trigger one or more workflow and manager their result.

> ## solution

Reviewer: ```suggestion
## Examples
```

> `WorkflowTemplateRef` can be used to create workflow with in two line.

Reviewer: ```suggestion
You can use `workflowTemplateRef` to trigger a workflow inline
```

1. Define your workflow as a `workflowtemplate`.
E.g:
```yaml
apiVersion: argoproj.io/v1alpha1
kind: WorkflowTemplate
metadata:
  name: workflow-template-whalesay-template
spec:
  entrypoint: whalesay-template
  templates:
  - name: whalesay-template
    inputs:
      parameters:
      - name: message
    container:
      image: docker/whalesay
      command: [cowsay]
      args: ["{{inputs.parameters.message}}"]

> ---

Reviewer: Please put each object on its own codeblock (do not use `---` to separate), it's easier to read that way
